autoload/lsp/omni.vim

```diff
@@ -63,7 +63,7 @@ function! lsp#omni#complete(findstart, base) abort
             while s:completion['status'] is# s:completion_status_pending && !complete_check()
                 sleep 10m
             endwhile
-            let s:completion['matches'] = filter(s:completion['matches'], {_, match -> match['word'] =~ '^' . a:base})
+            let s:completion['matches'] = filter(s:completion['matches'], {_, match -> stridx(match['word'], a:base) == 0})
```

Comment:
should we do a case insensitve `stridx` by calling `toUpper`? But not sure if people will like it. Might need to add in under flag.

Reply:
Actually I think the old `match` does case insenstive. so this could be breaking. If people complain we could add the flag back.

Reply:
Changed to check with tolower. (I think the check by regular expression is dangerous)